src/Views/ProgressView.vala

```diff
@@ -39,18 +44,35 @@ public class Sideload.ProgressView : AbstractView {
         }
     }
 
+    public ProgressType view_type { get; construct; }
+
+    public ProgressView (ProgressType type) {
+        Object (view_type: type);
+    }
+
     construct {
-        secondary_label.use_markup = true;
-        secondary_label.label = _("Preparing…");
+        if (view_type == ProgressType.REF_INSTALL) {
+            secondary_label.use_markup = true;
+            secondary_label.label = _("Preparing…");
+        } else if (view_type == ProgressType.BUNDLE_INSTALL) {
+            secondary_label.label = _("The application is being installed, it may take a while…");
```

Comment:
Do we know the size of the app in this context, at least, so we could say, “Installing 123.45 MB. Unable to estimate time remaining.” or something?

```suggestion
            secondary_label.label = _("Unable to estimate time remaining.");
```